### PR TITLE
UNFILED: add test for structtured plugin

### PR DESCRIPTION
**Pull request by jeremy-flieder**

### CODE REVIEW

File: Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java [hunk 1]
```diff
@@ -81,7 +82,173 @@ public void setUp(@Optional("http://localhost:8665/rapture") String url, @Option
         pluginApi = helper.getPluginApi();
         installedSet = new HashSet<String>();
     }
+    
+    @Test(groups={"plugin","nightly"})
+    public void testInstallStructuredPlugin () {
+    	String pluginName="teststruct";
+    	String description="Test structured plugin";
+    	String zipFileName="teststructcreate.zip";
```
Comment (scarabus) [inline Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java]: is teststructcreate.zip checked in?

File: Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java [hunk 2]
```diff
@@ -81,7 +82,173 @@ public void setUp(@Optional("http://localhost:8665/rapture") String url, @Option
...
+        Map<String, PluginTransportItem> payload = Maps.newHashMap();
+        Set <String> itemSet=new HashSet<String>();
+        for (PluginSandboxItem item : sandbox.getItems(null)) {
+            try {
+                PluginTransportItem payloadItem = item.makeTransportItem();
+                payload.put(item.getURI().toString(), payloadItem);
+                
+                itemSet.add(item.getURI().toString());
+            } catch (Exception ex) {
+                Reporter.log("Exception creating plugin " +ex.getMessage(),true);
```
Comment (scarabus) [inline Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java]: Is it OK that this isn't a failure?

File: Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java [hunk 3]
```diff
@@ -81,7 +82,173 @@ public void setUp(@Optional("http://localhost:8665/rapture") String url, @Option
...
+        entries = orgZipFile.entries();
+        while (entries.hasMoreElements()) {
+            ZipEntry entry = entries.nextElement();
+            if (entry.isDirectory()) {
+                continue;
+            }
+            try {
+            	sandbox.makeItemFromZipEntry(orgZipFile, entry);
+            } catch (Exception e) {
+            	Reporter.log("Error making sandbox item",true);
```
Comment (scarabus) [inline Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java]: Is this not a failure? At least log the message?

File: Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java (hunk 1 above)
Comment (jeremy-flieder) [inline Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java]: Yes it's in:

https://github.com/RapturePlatform/Rapture/blob/master/Apps/RaptureIntTests/src/test/resources/plugin/nightly/teststructcreate.zip

File: Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java (hunk 2 above)
Comment (jeremy-flieder) [inline Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java]: Put in better checking

File: Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java (hunk 3 above)
Comment (jeremy-flieder) [inline Apps/RaptureIntTests/src/test/java/rapture/plugin/PluginApiTest.java]: Put in better checking

**Pull request merged**